tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Builders/OrderSummaryDataBuilder.cs

```diff
@@ -0,0 +1,143 @@
+                        .WithOrderId(orderId)
+                        .WithOdsCode("Ods1")
+                        .WithCatalogueItemName("Order Item 1 ")
+                        .WithCatalogueItemType(CatalogueItemType.Solution)
+                        .Build())
+                .AddOrderItemEntity(
+                    OrderItemEntityBuilder.Create()
+                        .WithOrderId(orderId)
+                        .WithOdsCode("Ods2")
+                        .WithCatalogueItemName("Order Item 2 ")
+                        .WithCatalogueItemType(CatalogueItemType.AssociatedService)
+                        .Build())
+                .AddOrderItemEntity(
+                    OrderItemEntityBuilder.Create()
+                        .WithOrderId(orderId)
+                        .WithOdsCode("Ods3")
+                        .WithCatalogueItemName("Order Item 3 ")
+                        .WithCatalogueItemType(CatalogueItemType.AdditionalService)
+                        .Build());
+
+        public OrderSummaryDataBuilder RemoveOrders()
+        {
+            _OrderEntities = new List<OrderEntity>();
+            return this;
+        }
+
+        public OrderSummaryDataBuilder AddOrderEntity(OrderEntity orders)
+        {
+            _OrderEntities.Add(orders);
+            return this;
+        }
+
+        public OrderSummaryDataBuilder AddServiceRecipientEntity(ServiceRecipientEntity serviceRecipient)
+        {
+            _serviceRecipientEntites.Add(serviceRecipient);
+            return this;
+        }
+
+        public OrderSummaryDataBuilder RemoveServiceRecipients()
+        {
+            _serviceRecipientEntites = new List<ServiceRecipientEntity>();
+            return this;
+        }
+
+        public OrderSummaryDataBuilder WithOrderItemEntityList(List<OrderItemEntity> orderItems)
+        {
+            _OrderItemEntities = orderItems ?? new List<OrderItemEntity>();
+            return this;
+        }
+
+        public OrderSummaryDataBuilder RemoveAssociatedServices()
+        {
+            _OrderItemEntities.RemoveAll(oi => oi.CatalogueItemType == CatalogueItemType.AssociatedService);
+            return this;
+        }
+
+        public OrderSummaryDataBuilder RemoveAdditionalServices()
+        {
+            _OrderItemEntities.RemoveAll(oi => oi.CatalogueItemType == CatalogueItemType.AdditionalService);
+            return this;
+        }
+
+        public OrderSummaryDataBuilder RemoveSolutionsServices()
+        {
+            _OrderItemEntities.RemoveAll(oi => oi.CatalogueItemType == CatalogueItemType.Solution);
+            return this;
+        }
+
+        public OrderSummaryDataBuilder AddOrderItemEntity(OrderItemEntity orderItem)
+        {
+            _OrderItemEntities.Add(orderItem);
+            return this;
+        }
+
+        public OrderSummaryDataBuilder Build()
+        {
+            return this;
+        }
+
+        public async Task InsertAsync(string connectionString)
```

Comment:
Please review the use of two responsibilities. A builder should just know about building an object. The other object can deal with the insertion of the data.

Reply:
Added second class containing the Insert method